File: 2016-07-06_rxjava_tips/src/main/java/com/gdgand/rxjava/tips/MainActivity.java
package com.gdgand.rxjava.tips;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.gdgand.rxjava.tips.example.FirstExampleActivity;
import com.gdgand.rxjava.tips.example.SecondExampleActivity;
import com.gdgand.rxjava.tips.example.ThirdExampleActivity;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
    }

    @OnClick(R.id.button1)
    public void moveToExample1 () {
        startActivity(new Intent(this, FirstExampleActivity.class));
    }

    @OnClick(R.id.button2)
    public void moveToExample2 () {
        startActivity(new Intent(this, SecondExampleActivity.class));
    }

    @OnClick(R.id.button3)
    public void moveToExample3 () {
        startActivity(new Intent(this, ThirdExampleActivity.class));
    }
}

File: 2016-07-12_rxjava_in_action/src/main/kotlin/com/androidhuman/example/rxjavainaction/MainActivity.kt
package com.androidhuman.example.rxjavainaction

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val btnViewPager by lazy {
        findViewById(R.id.btn_activity_main_view_pager) as Button
    }

    private val btnViewPagerFlipInterval by lazy {
        findViewById(R.id.btn_activity_main_view_pager_with_interval) as Button
    }

    private val btnViewPagerFlipIntervalDragHandling by lazy {
        findViewById(R.id.btn_activity_main_view_pager_interval_drag_handling) as Button
    }

    private val btnVisibilityEvent by lazy {
        findViewById(R.id.btn_activity_main_visibility_event) as Button
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnViewPager.setOnClickListener {
            startActivity(Intent(this@MainActivity, ViewPagerActivity::class.java))
        }

        btnViewPagerFlipInterval.setOnClickListener {
            startActivity(Intent(this@MainActivity, ViewPagerFlipIntervalActivity::class.java))
        }

        btnViewPagerFlipIntervalDragHandling.setOnClickListener {
            startActivity(Intent(this@MainActivity,
                    ViewPagerFlipIntervalDragHandlingActivity::class.java))
        }

        btnVisibilityEvent.setOnClickListener {
            startActivity(Intent(this@MainActivity, VisibilityEventActivity::class.java))
        }
    }
}